File: Ejercicio 1/banco.java
abstract class Cuenta {
    protected double saldo;

    public Cuenta(double saldo) {
        this.saldo = saldo;
    }

    public abstract double calcularInteres();
}

class CuentaAhorro extends Cuenta {
    public CuentaAhorro(double saldo) {
        super(saldo);
    }

    public double calcularInteres() {
        return saldo * 0.02;
    }
}

class CuentaCorriente extends Cuenta {
    public CuentaCorriente(double saldo) {
        super(saldo);
    }

    public double calcularInteres() {
        return saldo * 0.01;
    }
}

class CuentaEmpresarial extends Cuenta {
    public CuentaEmpresarial(double saldo) {
        super(saldo);
    }

    public double calcularInteres() {
        return saldo * 0.03;
    }
}

interface Procesable {
    void procesar(double monto);
}

class Transferencia implements Procesable {
    public void procesar(double monto) {
        System.out.println("Transferencia procesada por " + monto);
    }
}

public class Main {
    public static void main(String[] args) {
        Cuenta[] cuentas = {
            new CuentaAhorro(1000),
            new CuentaCorriente(2000),
            new CuentaEmpresarial(3000)
        };

        for (Cuenta c : cuentas) {
            System.out.println("Interés: " + c.calcularInteres());
        }
    }
}

File: Ejercicio 1/banco.py
from abc import ABC, abstractmethod


class Cuenta(ABC):
    def __init__(self, saldo):
        self.saldo = saldo

    @abstractmethod
    def calcular_interes(self):
        pass


class CuentaAhorro(Cuenta):
    def calcular_interes(self):
        return self.saldo * 0.02

class CuentaCorriente(Cuenta):
    def calcular_interes(self):
        return self.saldo * 0.01

class CuentaEmpresarial(Cuenta):
    def calcular_interes(self):
        return self.saldo * 0.03


class Procesable(ABC):
    @abstractmethod
    def procesar(self, monto):
        pass


class Transferencia(Procesable):
    def procesar(self, monto):
        print(f"Transferencia procesada por {monto}")


cuentas = [CuentaAhorro(1000), CuentaCorriente(2000), CuentaEmpresarial(3000)]

for c in cuentas:
    print("Interés:", c.calcular_interes())
